FC-Chess/FC-Chess.cpp

```diff
@@ -107,28 +105,271 @@ void FCC::Chess::checkMovesHorizontal()
+        posPossible[i][j] = true;
+        if (checkIsEnemyOnField(&i, &j)) break;
+        i++;
+        j--;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    for (; i >= 0 && j <= 7;)
+    {
+        if (checkIsFriendlyOnField(&i, &j)) break;
+        posPossible[i][j] = true;
+        if (checkIsEnemyOnField(&i, &j)) break;
+        i--;
+        j++;
+    }
+    
+    i = selX, j = selY;     // resetting variables
+    for (; i >= 0 && j >= 0;)
+    {
+        if (checkIsFriendlyOnField(&i, &j)) break;
+        posPossible[i][j] = true;
+        if (checkIsEnemyOnField(&i, &j)) break;
+        i--;
+        j--;
+    }
+}
+
+void FCC::Chess::checkMovesKnight()
+{
+    // creating variables for x and y
+    uint8_t i = selX, j = selY;
+
+    // checking possible moves
+    i += 1, j += 3;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i += 1, j -= 3;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+    
+    i = selX, j = selY;     // resetting variables
+    i -= 1, j += 3;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i -= 1, j -= 3;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i += 3, j += 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i += 3, j -= 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i -= 3, j += 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i -= 3, j -= 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+}
+
+void FCC::Chess::checkMovesKing()
+{
+    // creating variables for x and y
+    uint8_t i, j;
+
+    // checking possible moves
+    i = selX - 1, j = selY + 1;
+    for (; i <= selX + 1; i++)
+    {
+        if (checkIsPosInBounds(&i, &j))
+        {
+            if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+        }
     }
+
+    i = selX - 1, j = selY - 1;
+    for (; i <= selX + 1; i++)
+    {
+        if (checkIsPosInBounds(&i, &j))
+        {
+            if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+        }
+    }
+
+    i = selX - 1, j = selY;
+    if (checkIsPosInBounds(&i, &j))
```

Comment:
do
```cpp
if (checkIsPosInBounds(selX - 1, selY))
```

Reply:
applied in the same push as the comments above